modules/st2-elastic-field/directive.js

```diff
@@ -0,0 +1,177 @@
+            ],
+            styleString = '';
+         
+        angular.forEach(propertiesToCopy, function(property) {
+          styleString += property + ':' + computedStyle.getPropertyValue(property) + ';';
+        });
+        
+        target.setAttribute('style', styleString);
+        target.value = source.value;
+      }
+      
+      function adjust($textarea, $clone, force) {
+        if ($textarea.isBeingAdjusted && !force) {
+          return;
+        }
+        
+        var dimensions = getDimensions($textarea),
+            overflow = 'hidden',
+            cloneHeight;
+        
+        $textarea.isBeingAdjusted = true;
+        imitate($textarea[0], $clone[0]);
+        cloneHeight = $clone[0].scrollHeight;
+        
+        if (cloneHeight > dimensions.maxHeight) {
+          cloneHeight = dimensions.maxHeight;
+          overflow = 'scroll';
+        } else if (cloneHeight < dimensions.minHeight) {
+          cloneHeight = dimensions.minHeight;
+        }
+        
+        $textarea[0].style.overflowY = overflow;
+
+        if (dimensions.height !== cloneHeight) {
+          $textarea[0].style.height = cloneHeight + 'px';
+        }
+          
+        $textarea.isBeingAdjusted = false;
+      }
+      
+      return {
+        restrict: 'C',
+        require: 'ngModel',
+        link: function (scope, $textarea, attrs, ngModel) {
+          if ($textarea[0].nodeName !== 'TEXTAREA' || !window.getComputedStyle || $textarea.data('elasticField')) {
+            return;
+          }
+          
+          
+          var $window = angular.element(window),
+              textarea = $textarea[0],
+              $clone = angular.element('<textarea class="st2-elastic-field-clone" tabindex="-1">'),
+              clone = $clone[0],
+              initialStyle = window.getComputedStyle(textarea),
+              initialResize = initialStyle.getPropertyValue('resize'),
+              boundAdjust = adjust.bind(window, $textarea, $clone),
+              boundForcedAdjust = adjust.bind(window, $textarea, $clone, true);
+          
+          $textarea.minRows = window.parseInt(attrs.st2ElasticFieldMinRows, 10);
+          $textarea.maxRows = window.parseInt(attrs.st2ElasticFieldMaxRows, 10);
+          
+          $textarea
+            .css({
+              'resize': (initialResize === 'none' || initialResize === 'vertical') ? 'none' : 'horizontal'
+            })
+            .data('elasticField', true);
+          $clone.data('elasticField', true);
+          angular.element(document.body).append(clone);
+
+
+          textarea.addEventListener('input', boundAdjust, false);
+          
+          // Oninput in IE9 doesn't capture character deletion,
+          // so we use onkeyup to overcome this
+          if ('onpropertychange' in textarea && 'oninput' in textarea) {
+            textarea.addEventListener('keyup', boundAdjust, false);
+          }
+
+          $window.bind('resize', boundForcedAdjust);
```

Comment:
The way this directive works, this function would be useful only in case elements width would change because of window resize. In our case, form elements always reside within fixed width container by design, so it is not required, for now.

If you decide to keep it, that's fine, but then I wonder whether it might introduce a flood of recalcs and repaints? Should not we use requestAnimationFrame or something like that here? I'm usually avoiding direct DOM access so I don't have to think about that, letting angular handle that kind of problems for me.